### PR TITLE
DATA-1940 - API changes for datasets

### PR DESCRIPTION
**Pull request by tahiyasalam**
Directly pulled from [datasets scope](https://docs.google.com/document/d/1D6TGiQ9n1qip1jW2O_dg5ahNv__1O_pZztqwT3hF7fI/edit)

### CODE REVIEW

**Review by agreenb — CHANGES_REQUESTED** [review]
Awesome, super quick turnaround 🚀  Mostly nit requests for documentation changes

File: proto/viam/app/data/v1/data.proto [hunk 1]
```diff
@@ -104,6 +113,7 @@ message Filter {
   TagsFilter tags_filter = 14;
   // bbox_labels are used to match documents with the specified bounding box labels (using logical OR)
   repeated string bbox_labels = 15;
+  string dataset_id = 16;
```
[the remark this answers is not in the document]
Comment (tahiyasalam) [reply proto/viam/app/data/v1/data.proto:116]: I was thinking about this too, I think for now the concept of a single dataset makes sense.

File: proto/viam/app/data/v1/data.proto [hunk 2]
```diff
@@ -62,6 +62,15 @@ service DataService {
   // ConfigureDatabaseUser configures a database user for the Viam organization's MongoDB Atlas Data
   // Federation instance. It can also be used to reset the password of the existing database user.
   rpc ConfigureDatabaseUser(ConfigureDatabaseUserRequest) returns (ConfigureDatabaseUserResponse);
+
+  // AddBinaryDataToDatasetByIDs add the binary data to the dataset based on the given binary IDs.
```
[the remark this answers is not in the document]
Comment (tahiyasalam) [reply proto/viam/app/data/v1/data.proto]: Done

File: proto/viam/app/dataset/v1/dataset.proto [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  rpc RenameDataset(RenameDatasetRequest) returns (RenameDatasetResponse);
+
+  // ListDatasetsByOrganizationID lists all of the datasets for an organization.
+  rpc ListDatasetsByOrganizationID(ListDatasetsByOrganizationIDRequest) returns (ListDatasetsByOrganizationIDResponse);
+
+  // ListDatasetsByIDs lists all of the datasets specified by ID.
+  rpc ListDatasetsByIDs(ListDatasetsByIDsRequest) returns (ListDatasetsByIDsResponse);
+}
+
+// Dataset stores the metadata.
```
[the remark this answers is not in the document]
Comment (tahiyasalam) [reply proto/viam/app/dataset/v1/dataset.proto]: Done

**Review by agreenb — APPROVED** [review]
🚀🚀

**Pull request merged**